gapic_smoketest.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Gapic generator smoke tests.
+
+It generates GAPIC client libraries for a Google APIs in googleapis
+repo. The test will fail if any generation fails.
+"""
+
+import argparse
+import logging
+import os
+import subprocess
+import sys
+
+logger = logging.getLogger('smoketest')
+logger.setLevel(logging.INFO)
+
+test_languages = {
+    "java" : lambda api_name, api_version, log : runJavaTests(api_name, api_version, log),
+    #TODO: all other languages
+}
+
+test_apis = {
+    "pubsub" : ("v1", "google/pubsub/artman_pubsub.yaml"),
+    "logging" : ("v2", "google/logging/artman_logging.yaml"),
+    "speech" : ("v1", "google/cloud/speech/artman_speech_v1.yaml"),
+}
+
+def runJavaTests(api_name, api_version, log):
+    gapic_dir = "gapic-google-cloud-%s-%s" % (api_name, api_version)
+    # Run gradle test in the main java directory and also in the gapic directory.
+    cmds = ("%s/gradlew build test" % os.getcwd()).split()
+    exit_code = subprocess.call(cmds, stdout=log, stderr=log)
+    if exit_code:
+        return exit_code
+
+    cmds = ("%s/gradlew -p %s build test" % (os.getcwd(), gapic_dir)).split()
+    return subprocess.call(cmds, stdout=log, stderr=log)
+
+def run_smoke_test(api_name, language, root_dir, log, user_config):
+    log_file = _setup_logger(log)
+    failure = []
+    success = []
+    warning = []
+    (api_version, artman_yaml_path) = test_apis[api_name]
+    target = language + "_gapic"
+
+    # Generate client library for an API and language.
+    if _generate_artifact(artman_yaml_path,
+                          target,
+                          root_dir,
+                          log_file,
+                          user_config):
+        msg = 'Failed to generate %s of %s.' % (
+            target, artman_yaml_path)
+        failure.append(msg)
+        logger.info(msg)
+    else:
+        msg = 'Succeded to generate %s of %s.' % (
+            target, artman_yaml_path)
+        success.append(msg)
+        logger.info(msg)
+
+        # Test the generated client library.
+        logger.info("Starting testing of %s %s client library." %(language, api_name))
+        cwd = os.getcwd()
+        os.chdir("artman-genfiles/%s" % language)
+
+        if _test_artifact(test_languages[language], api_name, api_version, log_file):
+            msg = 'Failed to pass tests for %s library.' % (
```

Comment:
nit: s/Failed to pass tests for %s library./Tests failed for %s library.

Reply:
Done